docs/source/examples/Semantic_segmentation.rst

```diff
@@ -0,0 +1,246 @@
+
+.. role:: hidden
+    :class: hidden-section
+
+
+.. _intro:
+
+Semantic segmentation using Poutyne
+***********************************
+
+.. note::
+
+    - See the notebook `here <https://github.com/GRAAL-Research/poutyne/blob/master/examples/Semantic_segmentation.ipynb>`_
+    - Run in `Google Colab <https://colab.research.google.com/github/GRAAL-Research/poutyne/blob/master/examples/Semantic_segmentation.ipynb>`_
+
+Semantic segmentation refers to the process of linking each pixel in an image to a class label. We can think of semantic segmentation as image classification at a pixel level. The image below clarifies the definition of semantic segmentation.
+
+.. image:: /_static/img/semantic_segmentation.png
+
+In this example, we are going to use and train a convolutional Unet, in order to design a network for semantic segmentation. In other words, we formulate the task of semantic segmentation as an image translation problem. We download and use the VOCSegmentation 2007 dataset for this purpose.
+
+Unet (the network we use in this example) is a convolutional neural network, similar to the convolutional autoencoders. However, Unet takes advantage of shortcuts between the encoder (contraction path) and decoder (expanding path), which helps it handle the vanishing gradient problem. In the following sections, we are going to install and import the segmentation-models-PyTorch library, which contains different architectures of Unet.
+
+.. code-block:: python
+
+    %pip install segmentation-models-pytorch 
+    
+Let’s import all the needed packages.
+
+.. code-block:: python
+
+    import numpy as np
+    import matplotlib.pyplot as plt
+    import math
+    import os
+    import segmentation_models_pytorch as smp
+    import PIL
+    import cv2
+    
+    import torch
+    import torch.nn as nn
+    import torch.optim as optim
+    from torch.utils.data import DataLoader, Dataset, Subset
+    
+    import torchvision
+    import torchvision.models as models
+    import torchvision.transforms as tfms
+    import torchvision.datasets as datasets
+    from torchvision.utils import  make_grid, save_image
+    
+    from poutyne import Model, ModelCheckpoint, CSVLogger, set_seeds
+    
+    device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+    print('The current processor is ...', device)
+
+Training constants
+==================================
+
+.. code-block:: python
+
+    learning_rate = 0.0005
+    batch_size = 32
+    image_size = 224
+    num_epochs = 70
+    _mean = [0.485, 0.456, 0.406]  # mean of the imagenet dataset for normalizing 
+    _std = [0.229, 0.224, 0.225]  # std of the imagenet dataset for normalizing 
+    set_seeds(42)
```

Comment:
View comment about set_seed in other PR. Could be placed before the initialization of the network for instance.